modules/optimization/test/tests/optimizationreporter/constant_heat_source/gold/main_out_OptimizationReporter_0001.csv

```diff
@@ -1,5 +1,2 @@
-grad_parameter_results,measurement_time,measurement_values,measurement_xcoord,measurement_ycoord,measurement_zcoord,misfit_values,parameter_results,simulation_values
-0,0,226,0.2,0.2,0,0,1000,226
-0,0,254,0.8,0.6,0,0,0,254
-0,0,214,0.2,1.4,0,0,0,214
-0,0,146,0.8,1.8,0,0,0,146
+grad_parameter_results,parameter_results
+0,1000
```

Comment:
why is it taking less steps now? or at least saving less

Reply:
In the old system, the optimizationReporter created a lot of reporters even if
they were not needed or created in other objects also. This made outputting CSV files of the objectives value and the final
gradient very messy because you would need to output a lot of unneeded data
too. We refactored the system to make it more modular so now the Optimization
object only creates the reporters needed for the simulation.